examples/run/cpu/seeding_example.cpp

```diff
@@ -85,12 +109,26 @@ int seq_run(const traccc::seeding_input_config& i_cfg,
           ------------*/
 
         if (i_cfg.check_performance) {
-            traccc::event_map evt_map(event, i_cfg.detector_file,
-                                      common_opts.input_directory,
-                                      common_opts.input_directory, host_mr);
-            sd_performance_writer.write(vecmem::get_data(seeds),
-                                        vecmem::get_data(spacepoints_per_event),
-                                        evt_map);
+            if (i_cfg.run_detray_geometry == false) {
+
+                traccc::event_map evt_map(event, i_cfg.detector_file,
+                                          common_opts.input_directory,
+                                          common_opts.input_directory, host_mr);
+                sd_performance_writer.write(
+                    vecmem::get_data(seeds),
+                    vecmem::get_data(spacepoints_per_event), evt_map);
+
+            }
+
+            else if (i_cfg.run_detray_geometry == true) {
+                traccc::event_map2 evt_map(event, common_opts.input_directory,
+                                           common_opts.input_directory,
+                                           common_opts.input_directory);
+                sd_performance_writer.write(
+                    vecmem::get_data(seeds),
+                    vecmem::get_data(spacepoints_per_event), readOut.modules,
+                    evt_map);
+            }
```

Comment:
This verbosity in the code may not be a good thing in the end... :thinking: I would much rather just have:

```c++
if (i_cfg.run_detray_geometry) {
    traccc::event_map2 evt_map(event, common_opts.input_directory,
                                common_opts.input_directory,
                                common_opts.input_directory);
    sd_performance_writer.write(
        vecmem::get_data(seeds),
        vecmem::get_data(spacepoints_per_event), readOut.modules,
        evt_map);
} else {
    traccc::event_map evt_map(event, i_cfg.detector_file,
                              common_opts.input_directory,
                              common_opts.input_directory, host_mr);
    sd_performance_writer.write(
        vecmem::get_data(seeds),
        vecmem::get_data(spacepoints_per_event), evt_map);
}
```

I.e. I **think** the compiler can be smart enough to recognize that one of these options will always happen, but the way that you wrote it, I believe the compiler does need to  check the value of that variable twice...